devito/tools/utils.py

```diff
@@ -154,6 +154,17 @@ def filter_sorted(elements, key=None):
     return sorted(filter_ordered(elements, key=key), key=key)
 
 
+def filter_sparse(elements):
```

Comment:
no, there should actually be nothing Devito-related in ``tools``. So you can't use things such as ``is_DiscreteFunction``

perhaps you don't need a function for this -- with a list comprehension it should become a one liner

Reply:
Fixed now. In addition to putting the function the the incorrect place I overlooked something when I was initially thinking about this filter.

(The initial 'no' in your first sentence is redundant).